File: algorithms/math/fibonacci/javascript/fibonacci.js
function fibonacciRecursive(n) {
  if (!Number.isInteger(n) || n < 0) {
    throw new Error(
      `Fibonacci argument must be a non-negative integer, got ${n}`
    );
  }
  const calculate = (x) => {
    if (x <= 1) {
      return BigInt(x);
    }
    return calculate(x - 1) + calculate(x - 2);
  };
  return calculate(n);
}

File: algorithms/math/fibonacci/python/fibonacci.py
import functools


@functools.lru_cache(maxsize=None)
def fibonacci_recursive(n: int) -> int:
    if not isinstance(n, int):
        raise TypeError(
            f"fibonacci_recursive() argument must be an integer, got {type(n).__name__}"
        )
    if n < 0:
        raise ValueError(
            f"fibonacci_recursive() argument must be a non-negative integer, got {n}"
        )
    return n if n <= 1 else fibonacci_recursive(n - 1) + fibonacci_recursive(n - 2)
